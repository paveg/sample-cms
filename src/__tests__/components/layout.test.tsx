import * as React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../../components/layout';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

useRouter.mockImplementation(() => ({}));

describe('Layout', () => {
  describe('when include string children', () => {
    const component = renderer.create(<Layout>children</Layout>);
    it('renders correctly component', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when include children container', () => {
    const children = <div>children</div>;
    const component = renderer.create(<Layout>{children}</Layout>);
    it('renders correctly component', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
