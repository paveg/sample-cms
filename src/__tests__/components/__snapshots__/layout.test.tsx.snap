// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout when include children container renders correctly component 1`] = `
Array [
  <div
    className="MuiContainer-root sc-bdfBwQ MuiContainer-maxWidthLg"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation0"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <h4
              className="MuiTypography-root MuiTypography-h4"
            >
              BLOG TITLE
            </h4>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiButtonGroup-root"
              role="group"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                href="/"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiTypography-root MuiTypography-button MuiTypography-colorTextSecondary"
                  >
                    HOME
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                href="/about"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiTypography-root MuiTypography-button MuiTypography-colorTextSecondary"
                  >
                    ABOUT
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div>
      children
    </div>
  </div>,
  <div
    className="MuiContainer-root sc-gsTCUz hklsSN MuiContainer-maxWidthLg"
  >
    <ul>
      <li>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/contact"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              contact
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </li>
      <li>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/privacy_policy"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              privacy policy
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </li>
      <li>
        <span
          className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          Copyright 2020 Ryota Ikezawa.
        </span>
      </li>
    </ul>
  </div>,
]
`;

exports[`Layout when include string children renders correctly component 1`] = `
Array [
  <div
    className="MuiContainer-root sc-bdfBwQ MuiContainer-maxWidthLg"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation0"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <h4
              className="MuiTypography-root MuiTypography-h4"
            >
              BLOG TITLE
            </h4>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiButtonGroup-root"
              role="group"
            >
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                href="/"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiTypography-root MuiTypography-button MuiTypography-colorTextSecondary"
                  >
                    HOME
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
              <a
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-text"
                href="/about"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiTypography-root MuiTypography-button MuiTypography-colorTextSecondary"
                  >
                    ABOUT
                  </span>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    children
  </div>,
  <div
    className="MuiContainer-root sc-gsTCUz hklsSN MuiContainer-maxWidthLg"
  >
    <ul>
      <li>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/contact"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              contact
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </li>
      <li>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/privacy_policy"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
            >
              privacy policy
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </li>
      <li>
        <span
          className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          Copyright 2020 Ryota Ikezawa.
        </span>
      </li>
    </ul>
  </div>,
]
`;
